rules_poetry/poetry.bzl

```diff
@@ -7,10 +9,40 @@ POETRY_UNSAFE_PACKAGES = ["setuptools", "distribute", "pip", "wheel"]
 def _clean_name(name):
     return name.lower().replace("-", "_").replace(".", "_")
 
+def _get_python_interpreter_attr(rctx):
+    if rctx.attr.python_interpreter:
+        return rctx.attr.python_interpreter
+
+    if "win" in rctx.os.name:
+        return "python.exe"
+    else:
+        return "python3"
+
+def _resolve_python_interpreter(rctx):
+    python_interpreter = _get_python_interpreter_attr(rctx)
+
+    if rctx.attr.python_interpreter_target:
+        if rctx.attr.python_interpreter:
+            fail("python_interpreter_target and python_interpreter incompatible")
+
+        target = rctx.attr.python_interpreter_target
+        python_interpreter = rctx.path(target)
+
+        return python_interpreter
+
+    if "/" not in python_interpreter:
```

Comment:
it looks like this PR is adding some specialization for windows, is this check going to work there? I assume it would use a backslash but haven't used bazel on windows.

Reply:
I don't have access to a Windows machine so hard to say. 

Perhaps we can add Windows to the CI checks in a future PR?

Reply:
The logic is from here BTW 

https://github.com/bazelbuild/rules_python/blob/main/python/pip_install/pip_repository.bzl#L51-L68

Reply:
ok, this can be addressed in a different PR